source/utils/urlUtils.py

```diff
@@ -35,13 +35,18 @@ def isSamePageURL(targetURLOnPage: str, rootURL: str) -> bool:
 	if not targetURLOnPage or not rootURL:
 		return False
 
-	validSchemes = ("http", "https")
 	# Parse the URLs
 	parsedTargetURLOnPage: ParseResult = urlparse(targetURLOnPage)
-	if parsedTargetURLOnPage.scheme not in validSchemes:
-		return False
 	parsedRootURL: ParseResult = urlparse(rootURL)
-	if parsedRootURL.scheme not in validSchemes:
+	if (
+		parsedTargetURLOnPage.scheme == "file"
+		and parsedRootURL.scheme == "file"
+		and parsedTargetURLOnPage.path == parsedRootURL.path
+	):
+		return True
```

Comment:
Why can't `"file"` just be added to the list of valid schemes?

Reply:
We can't just add file to validSchemes since we check different things for file and for validSchemes (http and https).
For file, we want to check that target and root URLs have the same path. And for http and https, we check that URLs are the same, excluding scheme and fragment.

Reply:
> For file, we want to check that target and root URLs have the same path. And for http and https, we check that URLs are the same, excluding scheme and fragment.

Are these tests not equivalent? A file URL without scheme and fragment (by the slightly limited definition in use here) **is** just its path. See [RFC 8089](https://datatracker.ietf.org/doc/html/rfc8089) for the file sURI scheme.

The only hitch I can see is the slightly odd choice to consider URLs with different schemes to point to the same page, which they most certainly do not, so I don't see how that's really a blocker here, except insofar as it renders this entire feature flawed.

Reply:
Your reasoning makes sense. I'll test this in practice.